packages/preview/rose-pine/0.1.0/src/apply.typ

```diff
@@ -0,0 +1,35 @@
+#let apply-theme(content, theme-data) = [
+  #set page(fill: theme-data.base,)
+  #set text(fill: theme-data.text,)
+  #set table(fill: theme-data.surface, stroke: theme-data.highlight.high)
+  #set circle(stroke: theme-data.subtle, fill: theme-data.overlay)
+  #set ellipse(stroke: theme-data.subtle, fill: theme-data.overlay)
+  #set line(stroke: theme-data.subtle)
+  #set path(stroke: theme-data.subtle, fill: theme-data.overlay)
+  #set polygon(stroke: theme-data.subtle, fill: theme-data.overlay)
+  #set rect(stroke: theme-data.subtle, fill: theme-data.overlay)
+  #set square(stroke: theme-data.subtle, fill: theme-data.overlay)
+  #show link: it => [
+    #set text(fill: theme-data.iris)
+    #it
+  ]
+  #show ref: it => [
+    #set text(fill: theme-data.foam)
+    #it
+  ]
+  #show footnote: it => [
+    #set text(fill: theme-data.pine)
+    #it
+  ]
```

Comment:
sorry for not noticing this before, but I think this would be an improvement
```suggestion
  #show link: set text(fill: theme-data.iris)
  #show ref: set text(fill: theme-data.foam)
  #show footnote: set text(fill: theme-data.pine)
```
for instance for `link` it will prevent extra spaces around the link which might be unintended

Reply:
It's certainly better (I think originally I was thinking of adding more styling which is why I did it that way, but the end result is worse), but I actually don't see the extra spaces in rendered links? The spacing in PDFs generated with both variations in my quick testing appears to be the exact same number of pixels

Reply:
I meant that when you have a link directly after text with no space then this will keep it that way while your previous code will add an extra space (because there are spaces in the content block). Alternatively a code block instead of a content block would also have prevented the issue.